### ISSUE TITLE
Incorrect results in Decimal mode

### ISSUE THREAD

**Issue by InPermutation** [mrdudz/run6502#4]
I, sadly, ended up switching to `sim65` as @mrdudz [recommended](https://github.com/mrdudz/run6502/pull/3#issuecomment-1837216059) because of this bug: I can't get [Hexadecimal to Decimal Conversion](http://www.6502.org/source/integers/hex2dec.htm) to work correctly.

[Archive.zip](https://github.com/mrdudz/run6502/files/13587575/Archive.zip) contains an assembly-language program (`test.s`/`a.out`) that attempts to show the bug. It loads the 16-bit hexadecimal value `$0119` into `HTD_IN`, then attempts to convert it to 3-byte BCD, which I then print.

The test case should probably be simplified even further, but this is the context in which I ran across it.

r65.cfg:
```
MEMORY {
    ZP:         start = $0000, size = $0100, type = rw;
    RAM:        start = $0001, size = $F000, fill = no, type = rw;
    ROM:        start = $8000, size = $0, fill = yes, file = %O;
    VEC:        start = $FFFA, size = $0006, fill = no, type = ro;
}

SEGMENTS {
    ZEROPAGE:   load = ZP, type = zp;
    CODE:       load = RAM, type = rw;
}
```

Assemble:
```sh
ca65 test.s
ld65 test.o -o test.img -C r65.cfg
```

Run:
```sh
run6502 -M F000 -R 200 -l 200 a.out -X 0 | hexdump -C
```

Expected output:
```
00000000  19 01 0a 30 30 32 38 31  0a                       |...00281.|
```

Actual output:
```
00000000  19 01 0a 30 30 34 38 31  0a                       |...00481.|
```

**Comment by mrdudz** [comment]
Yeah, no surprises there :) I see printfs in the implementation of decimal mode for SBC, so perhaps i have noticed the problem back then. Or it was broken even in the original library :)
